tap-snapshots/test/typeEvaluate.test.ts.test.cjs

```diff
@@ -1001,6 +1001,12 @@ Object {
 exports[`test/typeEvaluate.test.ts TAP flatmap > must match snapshot 1`] = `
 Object {
   "of": Array [
+    Object {
+      "of": Object {
+        "type": "null",
+      },
+      "type": "array",
+    },
```

Comment:
makes sense! If you flatmap over an non-array it will be null

Reply:
Uuh. Yeah. The nulls should be inside the union 😛

Reply:
I realized that the `mapConcrete` is kinda risky and it's not obvious that if you have a union of other types that you can process them in "parallel" and then take the union of the result. I think I'll have to ponder this a bit more before we can merge it.